docs/docs/development/hardware-integration/dongle.mdx

```diff
@@ -0,0 +1,301 @@
+    defaultValue="layouts"
+    values={[
+      { label: "Layouts File", value: "layouts" },
+      { label: "Base File", value: "base" },
+      { label: "Imported", value: "imported" }
+    ]}
+  >
+<TabItem value="layouts">
+Copy the file into your dongle's folder. Then import the file, assign the matrix transform to it, and select it in the `chosen` node:
+
+```dts
+#import "my_keyboard-layouts.dtsi"
+
+&physical_layout0 {
+    transform = <&default_transform>;
+};
+
+/ {
+    chosen {
+        zmk,kscan = &mock_kscan;
+        zmk,physical-layout = &physical_layout0;
+    };
+};
+```
+
+Make sure that the labels `physical_layout0` and `default_transform` match those of the physical layout node and the matrix transform respectively.
+
+If there are multiple physical layouts in the file, you will need to copy over all of the remaining matrix transformations and assign them similarly.
+
+</TabItem>
+<TabItem value="base">
+Copy the physical layout node into your `my_keyboard_dongle.overlay` file. Make sure the matrix transform is assigned to it, and select it in the `chosen` node.
+
+```dts
+/ {
+    chosen {
+        zmk,kscan = &mock_kscan;
+        zmk,physical-layout = &physical_layout0;
+    };
+
+    physical_layout0: physical_layout_0 {
+        compatible = "zmk,physical-layout";
+        display-name = "Default Layout";
+        transform = <&default_transform>;
+        keys = //Long list of keys
+    };
+};
+```
+
+Make sure that the labels `physical_layout0` and `default_transform` match those of the physical layout node and the matrix transform respectively.
+
+If there are multiple physical layouts, you need only copy the one that applies to your keyboard.
+
+</TabItem>
+<TabItem value="imported">
+Include your desired layout at the top of the file. Then assign your matrix transform to it, taking care to match up the node labels correctly.
+
+```dts
+#include <layouts/<stuff>.dtsi>
+
+&physical_layout0 {
+    transform = <&default_transform>;
+};
+```
+
+Your node label will never be `physical_layout0`, make sure to adjust it accordingly.
+
+Finally, select your physical layout in the chosen node.
+
+```dts
+/ {
+    chosen {
+        zmk,kscan = &kscan0;
+        zmk,physical-layout = &physical_layout0;
+    };
+};
+```
+
+</TabItem>
+</Tabs>
+
+</TabItem>
+<TabItem value="nostudio">
+Add a barebones [physical layout](physical-layouts.md) without the `keys` property to your `my_keyboard_dongle.overlay` file.
+
+```dts
+/ {
+    physical_layout0: physical_layout_0 {
+        compatible = "zmk,physical-layout";
+        display-name = "Default Layout";
+        transform = <&default_transform>;
+    };
+};
+```
+
+Make sure that the `transform` property uses the same label as your matrix transform. Then select the physical layout in the same `chosen` node as previously written:
+
+```dts
+/ {
+    chosen {
+        zmk,kscan = &mock_kscan;
+        zmk,physical-layout = &physical_layout0;
+    };
+};
+```
```

Comment:
````suggestion
Make sure that the `transform` property uses the same label as your matrix transform. Then select the physical layout in the same `chosen` node that was previously created:

```dts
/ {
    chosen {
        zmk,kscan = &mock_kscan;
        zmk,physical-layout = &physical_layout0;  // New item added
    };
};
```
````
Maybe a comment like this?